org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/ProjectConfigurators.java

```diff
@@ -112,17 +140,38 @@ private String configuratorFailedMessage(InternalProjectConfigurator contributio
         return String.format("Project configurator '%s' failed to %s", contribution.getId(), operationName);
     }
 
-    private IResource markerLocation() {
+    private MarkerLocation markerLocation(Exception exception) {
         Optional<IProject> maybeProject = CorePlugin.workspaceOperations().findProjectByLocation(this.gradleBuild.getBuildConfig().getRootProjectDirectory());
         if (!maybeProject.isPresent()) {
-            return ResourcesPlugin.getWorkspace().getRoot();
+            return new MarkerLocation(ResourcesPlugin.getWorkspace().getRoot(), 0);
         }
         IProject project = maybeProject.get();
         try {
             PersistentModel persistentModel = CorePlugin.modelPersistence().loadModel(project);
-            return project.getFile(persistentModel.getbuildScriptPath());
+            IFile buildScript = project.getFile(persistentModel.getbuildScriptPath());
+            if (!buildScript.exists()) {
+                return new MarkerLocation(project, 0);
+            }
+
+            int lineNumber = 0;
+            if (exception instanceof UnresolvedDependencyException) {
+                String coordinates = ((UnresolvedDependencyException)exception).getCoordinates();
+                InputStream is = buildScript.getContents(true);
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
+                    String line;
+                    int num = 0;
+                    while((line = reader.readLine()) != null) {
+                        num++;
+                        if (line.contains(coordinates)) {
```

Comment:
It should be possible to try to find the line even if the dependency string is malformed by replacing the colons by spaces on the line. Don't how how hackish is this tho.

Reply:
It's already ridiculously hackish. The coordinates should come from Gradle. Your upcoming PR should make this colon replacement obsolete.

Reply:
I think the line number solution is not good enough. I'll create a simpler PR that puts the error marker on the build scripts without the line number and add that capability to the configurator API separately.